gene_finder.py

```diff
@@ -61,8 +87,25 @@ def rest_of_ORF(dna):
     'ATG'
     >>> rest_of_ORF("ATGAGATAGG")
     'ATGAGA'
+
+    >>> rest_of_ORF("TAG")
+    ''
+
     """
+    #stop codons are TAA, TAG, TGA
     # TODO: implement this
+    #stop_codons = ['TAA','TAG','TGA']
+    stop_codons = ['TAA','TAG','TGA']
+
+    for i in range (0, len(dna)-1,3): # move though the strand every three bacepairs
+
+        if ( len(dna) - i >=3):       # only check for stop if there are three or more bace pairs left
+            if dna[i:i+3] in stop_codons:   #stop if you find a stop codon
+
+                return dna[0:i]         #return dna before the codon
+
```

Comment:
I like your inline comments. But you don't need to write them for every line if you feel like you are spending too much time explaining trivial things into words.